pkg/installmanager/installmanager.go

```diff
@@ -1810,3 +1820,62 @@ func (m *InstallManager) overrideCredentialsModeOnInstallConfig(credentialsMode
 
 	return nil
 }
+
+func (m *InstallManager) patchAzureOverrideCreds() error {
+	overrideCredsPath := filepath.Join(m.WorkDir, "manifests", overrideCredsYAML)
+	overrideCredsBytes, err := ioutil.ReadFile(overrideCredsPath)
+	if err != nil {
+		return errors.Wrap(err, "error reading 99_cloud-creds-secret.yaml")
+	}
+	c, err := yamlutils.Decode(overrideCredsBytes)
+	if err != nil {
+		return errors.Wrap(err, "unable to decode 99_cloud-creds-secret.yaml content")
+	}
+	if isAzureCreds, _ := yamlutils.Test(c, "/metadata/name", "azure-credentials"); !isAzureCreds {
+		return errors.Wrap(err, "wrong input, secret name is not azure-credentials in 99_cloud-creds-secret.yaml")
+	}
+	region, err := exec.Command("bash", "-c", fmt.Sprintf("grep %s %s |awk -F ': ' '{print $2}'", "region", filepath.Join(m.WorkDir, "manifests", clusterConfigYAML))).Output()
+	if err != nil {
+		return errors.Wrap(err, "error to grep region from manifests")
+	}
+	infraName, err := exec.Command("bash", "-c", fmt.Sprintf("grep %s %s |awk -F ': ' '{print $2}'", "infrastructureName", filepath.Join(m.WorkDir, "manifests", clusterInfraConfigYAML))).Output()
+	if err != nil {
+		return errors.Wrap(err, "error to grep infrastructureName from manifests")
+	}
+	resourceGroupName, err := exec.Command("bash", "-c", fmt.Sprintf("grep %s %s |awk -F ': ' '{print $2}'", "resourceGroupName", filepath.Join(m.WorkDir, "manifests", clusterInfraConfigYAML))).Output()
+	if err != nil {
+		return errors.Wrap(err, "error to grep resourceGroupName from manifests")
+	}
+	regionBase64 := interface{}(base64.RawStdEncoding.EncodeToString(region))
+	infraNameBase64 := interface{}(base64.RawStdEncoding.EncodeToString(infraName))
+	resourceGroupNameBase64 := interface{}(base64.RawStdEncoding.EncodeToString(resourceGroupName))
```

Comment:
Are these values really supposed to be base64-encoded?

Reply:
Yes, these are data in secret which shall be base64

Reply:
Are the region, infrastructureName, and resourceGroupName also secret data? If so, why didn't they need to be decoded above? Anyway, I guess this will shake out in testing.

Reply:
These are raw data in the manifests file so need base64 encoding before fill up.